### ISSUE TITLE
Version from stage tags (alpha/beta/rc) not recognized

### ISSUE THREAD

**Issue by MSLeiter** [mtkennerly/poetry-dynamic-versioning#76]
I have poetry-dynamic-versioning working overall. However, when I add staging tags to the repo, they are ignored. For example, `v1.0.0-rc-1` or `v0.5.0-beta-0`.

## Detailed scenarios
- Add `v1.0.0` tag to latest commit, and run poetry-dynamic-versioning.exe
  - the version is updated to `1.0.0`
- However, add a `v1.0.0-rc-1` tag to the latest commit, and the script will not update the version.
  - pdv reports an older commit version with a `.postNN.dev0+<hash>` version (same as no tag)
  - Running dunamai.Version.from_git() with this tag will return a version of `1.0.0rc1`, so dunamai is recognizing the pattern.

It's not clear what I need to do to enable pdv to use the staging. I'd like to tag my feature branches with a specific alpha, beta, or rc tag when building pre-release deployments.

It's not clear from the documentation how to do this (do I need to change the default pattern property? It looks like the stage info is already in there...)

My relevant pyproject.toml sections:
```
[tool.poetry-dynamic-versioning]
enable = true
vcs = 'git'

[tool.poetry-dynamic-versioning.substitution]
files = ["src/*/__version__.py"]

[build-system]
requires = ["poetry-core>=1.0.0", "poetry-dynamic-versioning"]
build-backend = "poetry.core.masonry.api"
```

**Comment by mtkennerly** [comment]
Thanks for pointing this out. The issue is that Dunamai and poetry-dynamic-versioning have their own copies of the default pattern, and they got out of sync. I'll release an update shortly to fix this.

**Comment by MSLeiter** [comment]
Great - is there a version of the pattern I can hardcode in the pyproject.toml to handle this for the moment? I'm sure I could figure out the regex eventually, but if there is already a pattern that will work...  :)

Thanks.

**Comment by mtkennerly** [comment]
Sure! Here's the current latest in Dunamai: https://github.com/mtkennerly/dunamai/blob/b6165f5529147db401013fae7429388cbd4677dd/dunamai/__init__.py#L37-L42 . You'd just need to escape the backslashes to make it TOML-friendly.

**Comment by mtkennerly** [comment]
This fix is included in both v0.14.1 (for Poetry 1.1.x) and v1.0.0b2 (for Poetry 1.2.x).
